### ISSUE TITLE
`flutter upgrade` shows version output from the old version, not new version

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter#22254]
Experienced this while upgrading on a machine I havn't used for a while:

```
$ flutter upgrade
Downloading Dart SDK from Flutter engine 1ed25ca7b7e3e3e8047df050bba4174074c9b336...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 58.6M  100 58.6M    0     0  9822k      0  0:00:06  0:00:06 --:--:-- 11.0M
Building flutter tool...
Upgrading Flutter from /Users/mit/dev/flutter...
[..].
Downloading ios-release tools...

Flutter 0.5.1 • channel beta • https://github.com/flutter/flutter.git
Framework • revision c7ea3ca377 (4 months ago) • 2018-09-07 12:33:05 -0700
Engine • revision 58a1894a1c
Tools • Dart 2.0.0-dev.58.0.flutter-f981f09760

Running flutter doctor...
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v0.8.2, on Mac OS X 10.13.6 17G65, locale en-DK)
[✓] Android toolchain - develop for Android devices (Android SDK 28.0.0)
[✓] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
[✓] Android Studio (version 3.1)
[✓] VS Code (version 1.26.1)
[!] Connected devices
    ! No devices available
```

While the doctor output at the bottom does list 0.8.2, the version output just a few lines above it still lists 0.5.1. I find that pretty confusing.

**Comment by Hixie** [comment]
This is still happening.
cc @jonahwilliams

**Comment by christopherfujino** [comment]
This was fixed by: https://github.com/flutter/flutter/pull/38325, which was published on v1.9.0. Note, you'll only get this fix if the version you're upgrading **from** is at least v1.9.0. In other words, if you upgrade from v1.7.8 to v1.12.13, you will still see the bug, but on the next upgrade you won't.
